### ISSUE TITLE
AWS IDs don't match

### ISSUE THREAD

**Issue by daveroberts** [hashicorp/terraform#3063]
This is my first encounter with Terraform, so it's possible this is my fault.

Created a VPC in Terraform.  Doing a `terraform show` gives the following information:

```
dave@archer:~/terraform$ terraform show
aws_vpc.default:
  id = vpc-02d32066
  cidr_block = 10.128.0.0/16
  default_network_acl_id = acl-f2b24c96
  default_security_group_id = sg-45f08522
  dhcp_options_id = dopt-927b84f7
  enable_dns_hostnames = true
  main_route_table_id = rtb-e3c36787
  tags.# = 1
  tags.Name = airpair-example
```

But when I log into the aws console, I see different values for these created resources:

```
vpc: vpc-7ed0a11b
network_acl: acl-76a7d113
security group id: sg-3f1fb45b
dhcp options: dopt-9ceb16f9
route table id: rtb-c96d14ac
```

My two tf files:

```
dave@archer:~/terraform$ cat variables.tf 
variable "access_key" {
  description = "AWS access key"
}

variable "secret_key" {
  description = "AWS secret access key"
}

variable "region" {
  desription = "AWS region to host your network"
  default = "us-east-1"
}

variable "vpc_cidr" {
  description = "CIDR for VPC"
  default = "10.128.0.0/16"
}

variable "public_subnet_cidr" {
  description = "CIDR for public subnet"
  default = "10.128.0.0/24"
}

variable "private_subnet_cidr" {
  description = "CIDR for private subnet"
  default = "10.128.1.0/24"
}

/* Ubuntu 14.04 amis by region */
variable "amis" {
  description = "Base AMI to launch the instances with"
  default = {
    us-west-1 = "ami-049d8641"
    us-east-1 = "ami-a6b8e7ce"
  }
}
```

```
dave@archer:~/terraform$ cat aws-vpc.tf 
/* Setup our aws provider */
provider "aws" {
  access_key = "${var.access_key}"
  secret_key = "${var.secret_key}"
  region = "${var.region}"
}

/* Define our vpc */
resource "aws_vpc" "default" {
  cidr_block = "${var.vpc_cidr}"
  enable_dns_hostnames = true
  tags {
    Name = "airpair-example"
  }
}
```

**Comment by daveroberts** [comment]
I deleted the state files from my local drive, and used the AWS console to delete all my resources.  I deleted my access key and regenerated it.  When I received a new access key and secret, the secret was much longer than before (my credentials were ancient, and I suspect something changed).  I changed my credentials in terraform.tfvars and performed an apply.  All the resources were created according to terraform.

However, when I check the AWS console online, it's showing that I have NO resources.  This could easily be a bug on Amazon's side of the house, but I'm sceptical of what's actually happening under the hood.
